### PR TITLE
Styling

### PR DESCRIPTION
**Pull request by Blemming**
## Pretty big boy
1. Added a new theme for the project using vuetify blueprints and a bit of custom css
2. Added a generic filter component for style consistency
3. Refactored a lot of the code for the filters to simpyfy it and make it more readable and maintainable
4. Wrote tests for filters changed to query param

### CODE REVIEW

File: src/components/filter_drawer/filters/boss_filters/BossFilterHolder.vue [hunk 1]
```diff
@@ -11,54 +11,45 @@ import { useFilterStore } from '@/store/FilterStore'
...
+
+                  :disabled="bossFiltersCount < 1"
+                  @click="filterStore.CLEAR_BOSS_FILTERS()"
+                >
+                  Clear
+                  <!-- <v-icon icon="mdi-window-close" /> -->
+                </v-btn>
               </template>
-              <p>Clear Active Boss Filters</p>
+              <p>Clear all map filters</p>
```
Comment (FloKaemmerer) [inline src/components/filter_drawer/filters/boss_filters/BossFilterHolder.vue:45]: copy paste: should be Boss Filters

File: src/components/filter_drawer/filters/divination_card_filters/DivinationCardFilterHolder.vue [hunk 2]
```diff
@@ -49,17 +32,20 @@ function clearActiveDivinationCardFilters() {
...
             </v-tooltip>
           </v-col>
         </v-fade-transition>
       </v-row>
+      <template #actions="{ expanded }">
+        <v-icon :icon="expanded ? 'mdi-menu-up' : 'mdi-menu-down'" />
+      </template>
     </v-expansion-panel-title>
-    <v-expansion-panel-text>
+    <v-expansion-panel-text class="">
```
Comment (FloKaemmerer) [inline src/components/filter_drawer/filters/divination_card_filters/DivinationCardFilterHolder.vue:48]: empty class?

File: src/components/filter_drawer/filters/map_filters/MapFilterHolder.vue [hunk 3]
```diff
@@ -14,56 +14,46 @@ import { useFilterStore } from '@/store/FilterStore'
...
             </v-tooltip>
           </v-col>
         </v-fade-transition>
       </v-row>
+      <template #actions="{ expanded }">
+        <v-icon :icon="expanded ? 'mdi-menu-up' : 'mdi-menu-down'" />
+      </template>
     </v-expansion-panel-title>
-    <v-expansion-panel-text>
+    <v-expansion-panel-text class="">
```
Comment (FloKaemmerer) [inline src/components/filter_drawer/filters/map_filters/MapFilterHolder.vue:56]: empty class?

**Review by FloKaemmerer — COMMENTED** [review]
Looks great!!

File: src/components/generics/GenericFilter.vue [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+const { rangeSlider } = defineProps<Props>()
+const emit = defineEmits(['update:checkbox', 'clickedCheckbox', 'update:rangeSlider'])
+
+const internalRangeSlider = toRef(rangeSlider)
+
+const debouncedFn = useDebounceFn(() => {
+  emit('update:rangeSlider', internalRangeSlider.value)
+}, 300)
```
[the remark this answers is not in the document]
Comment (Blemming) [reply src/components/generics/GenericFilter.vue:24]: just left the value copilot suggested 😂 thought it was ok, I think making it too short counter acts the debounce but I'll lower it

File: src/components/generics/GenericFilter.vue [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  rangeSliderMin?: number
+  rangeSliderMax?: number
+  rangeSliderLabelPrepend?: number | string
+  rangeSliderLabelAppend?: number | string
+}
+
+const { rangeSlider } = defineProps<Props>()
+const emit = defineEmits(['update:checkbox', 'clickedCheckbox', 'update:rangeSlider'])
+
+const internalRangeSlider = toRef(rangeSlider)
```
Comment (FloKaemmerer) [inline src/components/generics/GenericFilter.vue:20]: I encountered A Problem with the handling of the RangeSlider Values.
The InternalRangeSlider Values seem not to be updated when changing the corresponding Values in the FilterStore eg. when you "clear" the Filters.

you can reproduce it by activating a filter, slide to a value, clear the active filters and reactivate the Filter.
The Slider will show the previous Values but the Filter will apply the initial default values:
Screenshot 2023-08-04 144930

Comment (Blemming) [reply src/components/generics/GenericFilter.vue:20]: Nice catch, I'll look into it

Comment (Blemming) [reply src/components/generics/GenericFilter.vue:20]: Should deactivating a filter and reenabling it reset the filter also ?

Comment (FloKaemmerer) [reply src/components/generics/GenericFilter.vue:20]: Not sure about that one.

But I'd tend to reset it.

**Review by FloKaemmerer — APPROVED** [review]

**Pull request merged**